pkg/tsdb/influxdb/fsql/arrow.go

```diff
@@ -0,0 +1,277 @@
+	case arrow.INT32:
+		return newDataField[int32](f)
+	case arrow.INT64:
+		return newDataField[int64](f)
+	case arrow.BOOL:
+		return newDataField[bool](f)
+	case arrow.TIMESTAMP:
+		return newDataField[time.Time](f)
+	case arrow.DURATION:
+		return newDataField[int64](f)
+	default:
+		return newDataField[json.RawMessage](f)
+	}
+}
+
+func newDataField[T any](f arrow.Field) *data.Field {
+	if f.Nullable {
+		var s []*T
+		return data.NewField(f.Name, nil, s)
+	}
+	var s []T
+	return data.NewField(f.Name, nil, s)
+}
+
+// copyData copies the contents of an Arrow column into a Data Frame field.
+func copyData(field *data.Field, col arrow.Array) error {
+	defer func() {
+		if r := recover(); r != nil {
+			fmt.Println(fmt.Errorf("panic: %s %s", r, string(debug.Stack())))
+		}
+	}()
+
+	colData := col.Data()
+
+	switch col.DataType().ID() {
+	case arrow.TIMESTAMP:
+		v := array.NewTimestampData(colData)
+		for i := 0; i < v.Len(); i++ {
+			if field.Nullable() {
+				if v.IsNull(i) {
+					var t *time.Time
+					field.Append(t)
+					continue
+				}
+				t := v.Value(i).ToTime(arrow.Nanosecond)
+				field.Append(&t)
+				continue
+			}
+			field.Append(v.Value(i).ToTime(arrow.Nanosecond))
+		}
+	case arrow.DENSE_UNION:
+		v := array.NewDenseUnionData(colData)
+		for i := 0; i < v.Len(); i++ {
+			sc, err := scalar.GetScalar(v, i)
+			if err != nil {
+				return err
+			}
+			value := sc.(*scalar.DenseUnion).ChildValue()
+
+			var d any
+			switch value.DataType().ID() {
+			case arrow.STRING:
+				d = value.(*scalar.String).String()
+			case arrow.BOOL:
+				d = value.(*scalar.Boolean).Value
+			case arrow.INT32:
+				d = value.(*scalar.Int32).Value
+			case arrow.INT64:
+				d = value.(*scalar.Int64).Value
+			case arrow.LIST:
+				d = value.(*scalar.List).Value
+			default:
+				d = value.(*scalar.Null)
+			}
+			b, err := json.Marshal(d)
+			if err != nil {
+				return err
+			}
+			field.Append(json.RawMessage(b))
+		}
+	case arrow.STRING:
+		copyBasic[string](field, array.NewStringData(colData))
+	case arrow.UINT8:
+		copyBasic[uint8](field, array.NewUint8Data(colData))
+	case arrow.UINT16:
+		copyBasic[uint16](field, array.NewUint16Data(colData))
+	case arrow.UINT32:
+		copyBasic[uint32](field, array.NewUint32Data(colData))
+	case arrow.UINT64:
+		copyBasic[uint64](field, array.NewUint64Data(colData))
+	case arrow.INT8:
+		copyBasic[int8](field, array.NewInt8Data(colData))
+	case arrow.INT16:
+		copyBasic[int16](field, array.NewInt16Data(colData))
+	case arrow.INT32:
+		copyBasic[int32](field, array.NewInt32Data(colData))
+	case arrow.INT64:
+		copyBasic[int64](field, array.NewInt64Data(colData))
+	case arrow.FLOAT32:
+		copyBasic[float32](field, array.NewFloat32Data(colData))
+	case arrow.FLOAT64:
+		copyBasic[float64](field, array.NewFloat64Data(colData))
+	case arrow.BOOL:
+		copyBasic[bool](field, array.NewBooleanData(colData))
+	case arrow.DURATION:
+		copyBasic[int64](field, array.NewInt64Data(colData))
+	default:
```

Comment:
As memtioned above it would be good to handle `arrow.STRUCT` types with a `value` field and process them as if the column consists only of the `value`.

Reply:
@mhilton Thanks for your comments. Any suggestion about adding `arrow.STRUCT` case here? I couldn't manage to satisfy the type checker. 
```
case arrow.STRUCT:
	copyBasic[????](field, array.NewStructData(colData))
```

Reply:
> @mhilton Thanks for your comments. Any suggestion about adding `arrow.STRUCT` case here? I couldn't manage to satisfy the type checker.
> 
> ```
> case arrow.STRUCT:
> 	copyBasic[????](field, array.NewStructData(colData))
> ```

In that case I would leave it. The struct thing is a nice-to-have, but certainly not necessary and easy to work around with the SQL.